### PR TITLE
[DROOLS-4922] LinkageError: attempted duplicate class definition for materialized Lambda during incremental compilation

### PR DESCRIPTION
**Pull request by mariofusco**

### CODE REVIEW

File: drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/util/lambdareplace/MaterializedLambda.java [hunk 1]
```diff
@@ -107,10 +114,13 @@ void setMethodParameter(MethodDeclaration methodDeclaration) {
     }
 
     private EnumDeclaration create(CompilationUnit compilationUnit) {
-        EnumDeclaration lambdaClass = compilationUnit.addEnum(className);
+        EnumDeclaration lambdaClass = compilationUnit.addEnum(temporaryClassName);
```
Comment (hellowdan) [inline drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/util/lambdareplace/MaterializedLambda.java:117]: Is there any possibility of this variable [temporaryClassName] has no value sometimes? Is it a problem?

File: drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/util/lambdareplace/MaterializedLambdaConsequence.java [hunk 2]
```diff
@@ -17,10 +17,13 @@
 
 package org.drools.modelcompiler.util.lambdareplace;
 
+import com.github.javaparser.StaticJavaParser;
```
Comment (hellowdan) [inline drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/util/lambdareplace/MaterializedLambdaConsequence.java:20]: import is not being used

File: drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/util/lambdareplace/MaterializedLambdaConsequence.java [hunk 3]
```diff
@@ -17,10 +17,13 @@
...
+import com.github.javaparser.ast.CompilationUnit;
```
Comment (hellowdan) [inline drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/util/lambdareplace/MaterializedLambdaConsequence.java:21]: import is not being used

File: drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/util/lambdareplace/MaterializedLambdaConsequence.java [hunk 4]
```diff
@@ -17,10 +17,13 @@
...
 import com.github.javaparser.ast.Modifier;
 import com.github.javaparser.ast.NodeList;
 import com.github.javaparser.ast.body.EnumDeclaration;
 import com.github.javaparser.ast.body.MethodDeclaration;
+import com.github.javaparser.ast.expr.MethodCallExpr;
```
Comment (hellowdan) [inline drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/util/lambdareplace/MaterializedLambdaConsequence.java:26]: import is not being used

**Comment by mariofusco** [comment]
Replaced by https://github.com/kiegroup/drools/pull/2715

**Pull request closed without merging**